### PR TITLE
Asynchronous resize and memset in Radix sort

### PR DESCRIPTION
**Pull request by meistdan**
Adding async. versions for resize and memset to `GPUMemory` as sync. one caused issues in HIPRT.

### CODE REVIEW

File: Orochi/GpuMemory.h [hunk 1]
```diff
@@ -91,8 +91,27 @@ class GpuMemory final
 		*this = std::move( tmp );
 	}
 
+	void resizeAsync( const size_t new_size, const bool copy = false, oroStream stream = 0 ) noexcept
```
Comment (KaoCC) [inline Orochi/GpuMemory.h:98]: Could you please also add the doxy notes similar to the `resize` API ? 
I think it's very important to indicate that the copy might not be finished even if the function call returns.

Comment (KaoCC) [reply Orochi/GpuMemory.h:98]: Alternatively, you could choose to merge this together with the original `resize` function by providing an additional `bool async = false` parameter.

**Comment by KaoCC** [comment]
Could you please describe what the bug is so that we can be aware of the issue and avoid that in the future ?
Thank you.

**Comment by meistdan** [comment]
> Could you please describe what the bug is so that we can be aware of the issue and avoid that in the future ? Thank you.

You need to use async. functions with the user-provided stream instead of sync. functions with the default stream. Otherwise, the operations may be executed out of order.

**Review by KaoCC — APPROVED** [review]

**Pull request merged**